src/averageAges.js

```diff
@@ -1,61 +1,42 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(({ sex, died }) => century ? sex === 'm'
+    && Math.ceil(died / 100) === century : sex === 'm');
+
+  const brosReducing = men.reduce((sum, man) => sum + man.died - man.born, 0);
+
+  const calculatingTheAverageAgeForPeople = brosReducing / men.length;
```

Comment:
still not fixed. You can create separated function where you pass an array, then calculating sum inside and return value divided by array length